### ISSUE TITLE
Unable to load second-level of 1-M relationship using L.LoadXX helper methods

### ISSUE THREAD

**Issue by maurolscla** [volatiletech/sqlboiler#663]
If you're having a generation problem please answer these questions before submitting your issue. Thanks!

### What version of SQLBoiler are you using (`sqlboiler --version`)?
`SQLBoiler v3.6.1`

### What is your database and version (eg. Postgresql 10)
Postgresql 11

### If this happened at runtime what code produced the issue? (if not applicable leave blank)

```
package main

import (
	"context"
	"log"
	"sqlboiler_relationships_loading/src/models"

	uuid "github.com/gofrs/uuid"
	"github.com/jmoiron/sqlx"
	"github.com/volatiletech/sqlboiler/boil"
	"github.com/volatiletech/sqlboiler/drivers/sqlboiler-psql/driver"
	"github.com/volatiletech/sqlboiler/queries/qm"
)

func main() {
	queryString := driver.PSQLBuildQueryString("local", "local", "test", "localhost", 5432, "disable")
	db := sqlx.MustConnect("postgres", queryString)

	boil.SetDB(db)

	userID, _ := uuid.NewV4()
	ctx := context.Background()

	user := &models.User{ID: userID.String(), Name: "Mauro"}
	user.Insert(ctx, db, boil.Infer())

	videoID, _ := uuid.NewV4()

	video := &models.Video{ID: videoID.String(), Name: "Video", Description: "Description", UserID: userID.String()}

	err := video.Insert(ctx, db, boil.Infer())
	if err != nil {
		log.Fatal(err)
	}

	var tagID uuid.UUID

	for i := 0; i < 10; i++ {

		tagID, _ = uuid.NewV4()

		tag := &models.Tag{ID: tagID.String(), Name: "Tag", VideoID: videoID.String()}

		tag.Insert(ctx, db, boil.Infer())
	}

	user = nil

	user, _ = models.FindUser(ctx, db, userID.String())

	user.L.LoadVideos(ctx, db, true, user, qm.Load(qm.Rels(models.VideoRels.Tags)))

	if user.R.Videos[0].R.Tags == nil {
		println("Tags haven't been loaded")
	} else {
		println("Tags have been loaded")
	}

	user = nil

	user, _ = models.FindUser(ctx, db, userID.String())

	videos, err := user.Videos(qm.Load(qm.Rels(models.VideoRels.Tags))).All(ctx, db)

	user.R = user.R.NewStruct()

	user.R.Videos = videos

	if user.R.Videos[0].R.Tags == nil {
		println("Tags haven't been loaded")
	} else {
		println("Tags have been loaded")
	}

}
```

### Please provide a relevant database schema so we can replicate your issue (Provided you are comfortable sharing this)

```
CREATE TABLE "users" (
  "id" UUID,
  "name" TEXT NOT NULL,
  PRIMARY KEY ("id")
);

CREATE TABLE "videos" (
  "id" UUID,
  "user_id" UUID REFERENCES users(id) NOT NULL,
  "name" TEXT NOT NULL,
  "description" TEXT NOT NULL,
  PRIMARY KEY ("id")
);

CREATE TABLE "tags" (
  "id" UUID,
  "video_id" UUID REFERENCES videos(id) NOT NULL,
  "name" TEXT NOT NULL,
  PRIMARY KEY ("id")
);
```

### Further information. What did you do, what did you expect?
I have a previously loaded object which has not been eagerly loaded and I am trying to fetch two levels of dependencies using the L.LoadXXX helper method but I am not being successful. I am passing along the Load query mod to take care of that.
It can only load the second 1-M relationship level if I query it independently using `user.Videos()` (with the respective Load query mod for the tag relationship). But this forces me to manually build the R struct and assign the returning slice.

TL;DR: 
I am getting: 
`"Tags haven't been loaded"`
`"Tags have been loaded"`
I was expecting
`"Tags have been loaded"`
`"Tags have been loaded"`

am I missing something here?

**Comment by aarondl** [comment]
Hi there.

> user.L.LoadVideos(ctx, db, true, user, qm.Load(qm.Rels(models.VideoRels.Tags)))

Calling this method directly is not required. It's there to faciliate reflection magic (that's why its exported).

If you already have an object, then there's no ability to eager load (since the one part is already loaded!). Instead simple use a relationship helper like `.Tags()` and if you need to eager load from there you can.

Something like this should suffice:

```go
user, err := models.FindUser(...)

... := user.Videos(qm.Load(models.VideoRels.Tags))
```
